File: Back-End/src/main/java/nova/tech/com/backend/services/attribute/AttributeService.java
package nova.tech.com.backend.services.attribute;

import nova.tech.com.backend.dtos.request.AttributeRequest;
import nova.tech.com.backend.dtos.response.PageResponse;
import nova.tech.com.backend.models.Attribute;

import java.util.List;

public interface AttributeService {
    Attribute getAttributeById(long id);
    List<?> getAllAttributes(String categorySlug, String brandName);
    List<Attribute> getAttributeByCategory(String slug);
    void createAttribute(AttributeRequest attributeRequest);
    void updateAttribute(long id, AttributeRequest attributeRequest);
    void deleteAttribute(long id);
    PageResponse getAllAttributesWithPagination(int page, int limit);
}

File: Front-End/src/services/attribute.service.ts
import { del, get, post, put } from "@/services/axios";
import { AttributeRequest } from "@/types/request.type";
import { Attribute, Page } from "@/types/response.type";

export const getAttributeByCategory = async (slug: string): Promise<Attribute[]> => {
  const res = await get<Attribute[]>(`/attributes/category/${slug}`);
  return res.data;
}

export const createAttribute = async (attributeRequest: AttributeRequest) => {
  const res = await post("/attributes", attributeRequest);
  return res.data;
}

export const updateAttribute = async (id: number, attributeRequest: AttributeRequest) => {
  const res = await put(`/attributes/${id}`, attributeRequest);
  return res.data;
}

export const deleteAttribute = async (id: number) => {
  const res = await del(`/attributes/${id}`);
  return res.data;
}

export const getAllAttributesWithPagination = async (page: number, limit: number): Promise<Page<Attribute>> => {
  const res = await get<Page<Attribute>>(`/attributes/pagination?page=${page}&limit=${limit}`);
  return res.data;
}

export const getAttributeById = async (id: number): Promise<Attribute> => {
  const res = await get<Attribute>(`/attributes/${id}`);
  return res.data;
}